controllers/storagecluster/topology.go

```diff
@@ -280,3 +293,83 @@ func (r *StorageClusterReconciler) reconcileNodeTopologyMap(sc *ocsv1.StorageClu
 
 	return nil
 }
+
+// nodesHaveIdenticalValuesForKeys will return true only if
+// a. all the nodes have all the keys in their map
+// b. for a given node, values for all the keys are the same
+func nodesHaveIdenticalValuesForKeys(nodes *corev1.NodeList, keys []string) bool {
+
+	var match = true
+	if len(keys) == 0 {
+		return match
+	}
+
+	for _, node := range nodes.Items {
+		v, found := node.Labels[keys[0]]
+		if !found {
+			match = false
+		}
+		for _, key := range keys {
+			vnext, found := node.Labels[key]
+			if !found {
+				match = false
+			}
+			if v != vnext {
+				match = false
+			}
+		}
+	}
+	return match
+}
+
+func filterDuplicateLabels(sc *ocsv1.StorageCluster, nodes *corev1.NodeList, topologyMap *ocsv1.NodeTopologyMap) {
```

Comment:
Pretty obvious by the function name, but it may help to have a short comment above it regardless.

Reply:
Done